### PR TITLE
arrow v1.3.0

### PR DESCRIPTION
**Pull request by ELundby45**
arrow v1.3.0

**Destination channel:**  defaults

### Links

- [PKG-5694](https://anaconda.atlassian.net/browse/PKG-5694) 
- [Upstream repository](https://github.com/arrow-py/arrow/tree/1.3.0)
- [Upstream changelog/diff](https://github.com/arrow-py/arrow/releases/tag/1.3.0)
- [pypoject.toml](https://github.com/arrow-py/arrow/blob/1.3.0/pyproject.toml)

### Explanation of changes:

- Bump version and SHA
- Update pinnings to match up-stream
- Add upstream patch to remove the unnecessary `types-python-dateutil` dependency

[PKG-5694]: https://anaconda.atlassian.net/browse/PKG-5694?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by SC426 — APPROVED** [review]

File: recipe/patches/01_remove_type_checker.txt [hunk 1]
```diff
@@ -0,0 +1,41 @@
+From 06e2acc756b7c209ffc0bb4151712a972c493fe0 Mon Sep 17 00:00:00 2001
```
Comment (danpetry) [inline recipe/patches/01_remove_type_checker.txt]: usually patches are .patch

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,41 +1,49 @@
...
     - pip
     - pytest
     - pytest-cov
     - pytest-mock
     # Pin `pytz` to match upstream. Prevents chance for version mismatch with `python-dateutils`
     - pytz ==2021.1
-    - simplejson >=3.*
+    - simplejson ==3.*
   commands:
     - python -m pip check
```
Comment (danpetry) [inline recipe/meta.yaml]:
```suggestion
    - pip check
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,41 +1,49 @@
...
     - cd tests && pytest --cov arrow
```
Comment (danpetry) [inline recipe/meta.yaml]: I think you might be able to do pytest --cov tests/arrow? not sure.

**Review by danpetry — APPROVED** [review]
few minor nits

**Pull request merged**